### PR TITLE
feat(application-generic,shared): card element dsl and zod fixes NV-8385

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

Rich Chat Editor - foundations, card element DSL and Zod schema

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

<h3>Greptile Summary</h3>

This PR introduces the foundational DSL for rich chat card elements, adding TypeScript types (`CardElement` and its children), a matching Zod validation schema, a `isMailyChatBody` format-sniffer, and a new `IS_CHAT_BLOCK_EDITOR_ENABLED` feature flag.

- **Card element DSL** (`card-element.types.ts` + `card-element.schema.ts`): defines a structured `card → children` hierarchy (text, image, divider, actions/link-button) with corresponding Zod schemas that use `.strict()` and a discriminated union on `type`.
- **`isMailyChatBody` utility** (`chat-body.ts`): sniffs whether a stored chat `body` string is a Maily/TipTap document `{ type: 'doc', content: [] }` vs. a plain string, enabling the server render path to branch without a flag.
- Both new modules include vitest unit tests covering round-trips, unknown types, missing required fields, and edge-case JSON inputs.

<h3>Confidence Score: 4/5</h3>

Safe to merge for foundational/flag-gated work, but the URL fields in the card schema should be hardened before the feature is enabled in production.

All three URL fields in the card Zod schema (`image.url`, `link-button.url`, `card.imageUrl`) accept any string, including `javascript:` protocol URIs, which would pass validation and be stored as-is. If any rendering path — in the Novu dashboard, a preview, or a provider integration — places these values into an `href` or `src` without additional sanitization, stored cross-site scripting becomes possible. The fix is straightforward (`.url()` validator), but the issue is present in the foundation layer that everything else will build on.

**Files Needing Attention:** libs/application-generic/src/schemas/chat/card-element.schema.ts — all three URL string fields need format validation before this schema is used in production flows.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libs/application-generic/src/schemas/chat/card-element.schema.ts | New Zod schema for the card DSL. Structure and discriminated union are correct, but all three URL fields use bare `z.string()` without `.url()`, which permits `javascript:` protocol URIs to pass validation and be stored. |
| packages/shared/src/types/card-element.types.ts | New TypeScript types for the card element DSL; accurately mirror the Zod schema and are well-documented. |
| packages/shared/src/utils/chat-body.ts | New `isMailyChatBody` format-sniffer; correctly handles null, arrays, malformed JSON, and non-doc shapes. Logic is sound. |
| libs/application-generic/src/schemas/chat/card-element.schema.spec.ts | Test coverage includes round-trip, unknown type rejection, missing required field, and type-compatibility checks. |
| packages/shared/src/utils/chat-body.spec.ts | Good unit tests covering all edge cases: plain string, valid doc, malformed JSON, null, array, and wrong-shape objects. |
| packages/shared/src/types/feature-flags.ts | Adds `IS_CHAT_BLOCK_EDITOR_ENABLED` feature flag with appropriate JSDoc comment. |
| libs/application-generic/src/index.ts | Adds re-export for new `./schemas/chat` module. |

</details>

<h3>Class Diagram</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
classDiagram
    class CardElement {
        +type: 'card'
        +title?: string
        +subtitle?: string
        +imageUrl?: string
        +children: CardElementChild[]
    }
    class CardElementChild {
        <<union>>
    }
    class CardElementTextElement {
        +type: 'text'
        +content: string
        +style?: 'plain' | 'bold' | 'muted'
    }
    class CardElementImageElement {
        +type: 'image'
        +url: string
        +alt?: string
    }
    class CardElementDividerElement {
        +type: 'divider'
    }
    class CardElementActionsElement {
        +type: 'actions'
        +children: CardElementLinkButtonElement[]
    }
    class CardElementLinkButtonElement {
        +type: 'link-button'
        +label: string
        +url: string
        +style?: 'primary' | 'danger' | 'default'
    }
    CardElement "1" --> "*" CardElementChild : children
    CardElementChild <|-- CardElementTextElement
    CardElementChild <|-- CardElementImageElement
    CardElementChild <|-- CardElementDividerElement
    CardElementChild <|-- CardElementActionsElement
    CardElementActionsElement "1" --> "1..*" CardElementLinkButtonElement : children
```

<a href="https://app.greptile.com/api/ide/cursor?prompt=%23%23%23%20Issue%201%0Alibs%2Fapplication-generic%2Fsrc%2Fschemas%2Fchat%2Fcard-element.schema.ts%3A11-32%0A**URL%20fields%20accept%20%60javascript%3A%60%20and%20other%20non-HTTP%20schemes**%0A%0AAll%20three%20URL%20fields%20%E2%80%94%20%60url%60%20in%20%60cardElementImageZodSchema%60%20%28line%2014%29%2C%20%60url%60%20in%20%60cardElementLinkButtonZodSchema%60%20%28line%2029%29%2C%20and%20%60imageUrl%60%20in%20%60cardElementZodSchema%60%20%28line%2053%29%20%E2%80%94%20use%20bare%20%60z.string%28%29%60%20with%20no%20format%20constraint.%20Any%20stored%20card%20that%20carries%20a%20%60javascript%3Aalert%281%29%60%20URL%20will%20pass%20schema%20validation.%20Even%20though%20rendering%20is%20delegated%20to%20Slack%2FTeams%2FTelegram%2FWhatsApp%2C%20the%20data%20is%20also%20likely%20surfaced%20in%20Novu's%20own%20dashboard%2C%20where%20a%20%60javascript%3A%60%20href%20on%20an%20anchor%20or%20a%20%60src%60%20on%20an%20image%20could%20execute%20in%20the%20browser.%0A%0AAdding%20%60.url%28%29%60%20%28or%20%60.url%28%7B%20message%3A%20'...'%20%7D%29%60%29%20narrows%20the%20accepted%20set%20to%20%60http%3A%2F%2F%60%20and%20%60https%3A%2F%2F%60%20schemes%20and%20rejects%20malformed%20values%20at%20ingestion%20time%2C%20before%20they%20are%20stored%20or%20forwarded%20to%20providers.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&pr=12088&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursorDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCursor.svg?v=6">Fix All in Cursor</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
### Issue 1
libs/application-generic/src/schemas/chat/card-element.schema.ts:11-32
**URL fields accept `javascript:` and other non-HTTP schemes**

All three URL fields — `url` in `cardElementImageZodSchema` (line 14), `url` in `cardElementLinkButtonZodSchema` (line 29), and `imageUrl` in `cardElementZodSchema` (line 53) — use bare `z.string()` with no format constraint. Any stored card that carries a `javascript:alert(1)` URL will pass schema validation. Even though rendering is delegated to Slack/Teams/Telegram/WhatsApp, the data is also likely surfaced in Novu's own dashboard, where a `javascript:` href on an anchor or a `src` on an image could execute in the browser.

Adding `.url()` (or `.url({ message: '...' })`) narrows the accepted set to `http://` and `https://` schemes and rejects malformed values at ingestion time, before they are stored or forwarded to providers.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
`````

</details>

<sub>Reviews (2): Last reviewed commit: ["Merge branch &#39;next&#39; into nv-8385-rich-ch..."](https://github.com/novuhq/novu/commit/4d1c094f4fb9d07c71f643f816ca00bd70b65f8b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=47052622)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: libs/application-generic/src/schemas/chat/card-element.schema.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  .strict();
+
+const cardElementLinkButtonZodSchema = z
+  .object({
+    type: z.literal('link-button'),
+    label: z.string(),
+    url: z.string(),
+    style: z.enum(['primary', 'danger', 'default']).optional(),
+  })
+  .strict();
```
[the remark this answers is not in the document]
Comment (LetItRock) [reply libs/application-generic/src/schemas/chat/card-element.schema.ts:32]: it's a string type because it could also store the variables
